src/sentry/event_manager.py

```diff
@@ -451,11 +453,32 @@ def use_rust_normalize(self):
         return ENABLE_RUST
 
     def normalize(self):
+        tags = {
+            'project_id': six.text_type(
+                self._project.id
+                if self._project and self.use_rust_normalize
```

Comment:
This `and self.use_rust_normalize` is there to not send every possible project ID, but only when we care about it (when the rust code runs). Anton brought this up because datadog gets slow with high-cardinality tags.